src/WebJobs.Script/Description/DotNet/CSharp/CSharpCompilationService.cs

```diff
@@ -65,7 +66,7 @@ private static string GetFunctionSource(FunctionMetadata functionMetadata)
                 code = File.ReadAllText(functionMetadata.ScriptFile);
             }
 
-            return code ?? string.Empty;
+            return Utility.RemoveUtf8ByteOrderMark(code ?? string.Empty);
```

Comment:
I was trying to keep only the symptom description in the issue to avoid confusion, but the root of the problem was when we had a mismatch between the encoding used in the code and how we built the Roslyn syntax tree, which would lead to different hashes between the actual file and what was in the PDBs.

With this change, we're always stripping the BOM if present and setting the encoding used to create the syntax tree to UTF8 without a BOM (the default), ensuring that we have a match.